lib/srv/db/postgres/objects.go

```diff
@@ -0,0 +1,130 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package postgres
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v4"
+	"github.com/sirupsen/logrus"
+
+	dbobjectv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobject/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/srv/db/common/databaseobjectimportrule"
+	"github.com/gravitational/teleport/lib/srv/db/objects"
+)
+
+// TODO(Tener): add e2e tests for objectFetcher.
+type objectFetcher struct {
+	cfg objects.ObjectFetcherConfig
+	db  types.Database
+}
+
+var _ objects.ObjectFetcher = (*objectFetcher)(nil)
+
+func NewObjectFetcher(ctx context.Context, db types.Database, cfg objects.ObjectFetcherConfig) (objects.ObjectFetcher, error) {
+	if db.GetAdminUser().DefaultDatabase == "" || db.GetAdminUser().Name == "" {
```

Comment:
hmm. `default_database` is not really required for Postgres auto-user provisioning, and not used at all for the current DAC implementation IIRC.

Maybe pick a database name from the non-wildcard rules? Or default to "postgres" database?

Reply:
Hmm, I'm not certain how safe that heuristic is, but I like the idea of reducing the configuration requirements. What about trying either `postgres` or `teleport-admin`?

Reply:
Let's do `postgres` so users don't have to create a new database. we can make the doc clear that either the user gives admin user permission to connect `postgres` or specify a default database.

Reply:
Makes sense, I'll do so.

Reply:
Done: the requirement for db name was dropped, we default to "postgres" in "get all db names" function.